### PR TITLE
docs: add sample usage of the FormArray API for reactive forms

### PR DESCRIPTION
**Pull request by maequise**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Missing a more concrete sample on how to use the FormArray API

Issue Number: #53705

## What is the new behavior?
Include a more concrete sample of how to use the FormArray API on the site into the guide of reactive forms

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jessicajaniuk** [comment]
@maequise Looks like your commit message is causing the lint to fail. It's very close though. Run `git commit --amend` and change your commit message to `docs: add sample usage of the FormArray API for reactive forms`, and then force push. That'll fix it.

**Comment by jessicajaniuk** [comment]
@dylhunn Can you review the content here?

File: adev/src/content/guide/forms/reactive-forms.md [hunk 1]
```diff
@@ -392,6 +392,192 @@ Initially, the form contains one `Alias` field. To add another field, click the
...
+
+<docs-workflow>
+<docs-step title="Define the FormGroup and template">
+
+```ts
+@Component({
+  selector: 'my-form-comp',
+  standalone: true,
+  imports: [ReactiveFormsModule]
+  //component declaration
```
Comment (sonukapoor) [inline adev/src/content/guide/forms/reactive-forms.md:414]: Do we need this comment?

Comment (maequise) [reply adev/src/content/guide/forms/reactive-forms.md:414]: This comment is not revelant, I certainly forgot to remove it. If need I can remove it and make a new push.

Comment (sonukapoor) [reply adev/src/content/guide/forms/reactive-forms.md:414]: Yes, remove anything that is not relevant.

File: adev/src/content/guide/forms/reactive-forms.md [hunk 2]
```diff
@@ -392,6 +392,192 @@ Initially, the form contains one `Alias` field. To add another field, click the
 
 </docs-workflow>
 
+## A more concrete usage of FormArray
+
+In a more concrete sample, we can imagine the following use case. You are application needs a basic form, with known information, but the user has the ability to add more or less sub-form.
```
Comment (sonukapoor) [inline adev/src/content/guide/forms/reactive-forms.md:397]:
```suggestion
In a more concrete sample, we can imagine the following use case. Your application needs a basic form, with known information, but the user has the ability to add more or less sub-form.
```

File: adev/src/content/guide/forms/reactive-forms.md [hunk 3]
```diff
@@ -392,6 +392,192 @@ Initially, the form contains one `Alias` field. To add another field, click the
...
+    }
+    
+    <button type="submit()">Submit</button>
+  </form>
+</div>
+```
+
+And in the template we have our principal form displayed and mapped, and a `@for` loop used to iterate over the `FormArray`.
+
+Tip: If you watch the HTML template, you will notice that the access to the `FormArray` is not from the `FormGroup` directly with the **get()** method. We will define a getter method.
```
Comment (sonukapoor) [inline adev/src/content/guide/forms/reactive-forms.md:480]:
```suggestion
Tip: If you watch the HTML template, you will notice that the access to the `FormArray` is not from the `FormGroup` but directly using the **get()** method. We will define a getter method.
```

File: adev/src/content/guide/forms/reactive-forms.md [hunk 4]
```diff
@@ -392,6 +392,192 @@ Initially, the form contains one `Alias` field. To add another field, click the
...
+export class FormArrayComp {
+  //boilerplate code
+  addEntry() : void {
+    this.form.push(this.fb.group({
+      subControlName: ['']
+    }));
+  }
+}
+```
+With this method, when the user click on the button, it is adding a new `FormGroup` in the `FormArray`. When adding a new entry, the template will be updated.
```
Comment (sonukapoor) [inline adev/src/content/guide/forms/reactive-forms.md:534]:
```suggestion
With this method, when the user clicks on the button, it is adding a new `FormGroup` in the `FormArray`. When adding a new entry, the template will be updated.
```

File: adev/src/content/guide/forms/reactive-forms.md [hunk 5]
```diff
@@ -392,6 +392,192 @@ Initially, the form contains one `Alias` field. To add another field, click the
...
+</docs-step>
+
+<docs-step title="Add possibility to remove an entry">
+
+We can add entries to the `FormArray` but we should be able to remove a specific entry. The `FormArray` has the **removeAt()** method that permits to delete a specific index of the `FormArray`.
```
Comment (sonukapoor) [inline adev/src/content/guide/forms/reactive-forms.md:539]:
```suggestion
We can now add entries to the `FormArray` but we should also be able to remove a specific entry. The `FormArray` has the **removeAt()** method that permits to delete a specific index of the `FormArray`. 
```

**Comment by maequise** [comment]
@sonukapoor  thank you for the review, I made the corrections (the last forced "push"  was because I missed one edit)

**Comment by MarkTechson** [comment]
I really appreciate this work but this should really be a tutorial or code example, not necessarily an additional guide. I would also add here that we should consider revamping this guide and converting it to an issue that we should track and keep track of this work.

@twerske can you chime in on this?

**Comment by maequise** [comment]
> I really appreciate this work but this should really be a tutorial or code example, not necessarily an additional guide. I would also add here that we should consider revamping this guide and converting it to an issue that we should track and keep track of this work.
> 
> @twerske can you chime in on this?

I was thinking earlier in the afternoon adding a tutorial on the suject.

**Comment by twerske** [comment]
> I really appreciate this work but this should really be a tutorial or code example, not necessarily an additional guide. I would also add here that we should consider revamping this guide and converting it to an issue that we should track and keep track of this work.
> 
> @twerske can you chime in on this?

I agree. This should be a tutorial not a guide since it's a walk through of a code example.

**Comment by maequise** [comment]
As suggested, I made a tutorial for this part and removed the guide part.

**Review by BojanKogoj — CHANGES_REQUESTED** [review]

**Pull request closed without merging**